src/main/java/chess/domain/square/Square.java

```diff
@@ -0,0 +1,66 @@
+package chess.domain.square;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public record Square(File file, Rank rank) {
+
+    public static Square from(String square) {
+        File file = File.from(square.substring(0, 1));
+        Rank rank = Rank.from(square.substring(1, 2));
+        return new Square(file, rank);
+    }
+
+    public List<Square> generatePath(Square target) {
+        List<Square> path = new ArrayList<>();
+
+        int vectorFile = target.signumFile(file);
+        int vectorRank = target.signumRank(rank);
+
+        Square current = this;
+        do {
+            current = current.addVector(vectorFile, vectorRank);
+            path.add(current);
+        } while (!target.equals(current));
+
+        return path;
+    }
+
+    private Square addVector(int vectorFile, int vectorRank) {
+        return new Square(
+                this.file.add(vectorFile),
+                this.rank.add(vectorRank));
```

Comment:
```suggestion
        return new Square(this.file.add(vectorFile), this.rank.add(vectorRank));
```
(반영하지 않으셔도 됩니다.) 한칸 내리는 것은 생각보다 많은 가독성 저해를 일으키니 가능하면 내리지 않는 것이 좋습니다. 현재 코드는 길이가 짧아 내릴 필요가 없어보이기도하구요 :)

Reply:
오리의 의견에 동의합니다. 반영하였습니다.😸